jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/udp/ssm/SsmReceiver.java

```diff
@@ -54,34 +42,20 @@ public void run() {
                 logger.debug("Waiting for UDP SSM packets...");
                 socket.receive(packet);
                 if (packet.getLength() > 0) {
-                    senderIp = packet.getAddress().getHostAddress();
-                    senderPort = packet.getPort();
-                    logger.debug("Packet received from {}:{}", senderIp, senderPort);
-
-                    // Create OdeMsgPayload and OdeLogMetadata objects and populate them
-                    OdeAsn1Payload ssmPayload = super.getPayloadHexString(packet, UperUtil.SupportedMessageTypes.SSM);
-                    if (ssmPayload == null)
-                        continue;
-                    OdeSsmMetadata ssmMetadata = new OdeSsmMetadata(ssmPayload);
-
-                    // Add header data for the decoding process
-                    ZonedDateTime utc = ZonedDateTime.now(ZoneOffset.UTC);
-                    String timestamp = utc.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"));
-                    ssmMetadata.setOdeReceivedAt(timestamp);
-
-                    ssmMetadata.setOriginIp(senderIp);
-                    ssmMetadata.setSsmSource(SsmSource.RSU);
-                    ssmMetadata.setRecordType(RecordType.ssmTx);
-                    ssmMetadata.setRecordGeneratedBy(GeneratedBy.RSU);
-                    ssmMetadata.setSecurityResultCode(SecurityResultCode.success);
-
-                    // Submit JSON to the OdeRawEncodedMessageJson Kafka Topic
-                    ssmPublisher.publish(JsonUtils.toJson(new OdeAsn1Data(ssmMetadata, ssmPayload), false),
-                        ssmPublisher.getOdeProperties().getKafkaTopicOdeRawEncodedSSMJson());
+                    
+                    String ssmJson = UdpHexDecoder.buildJsonSsmFromPacket(packet);
+
+                    if(ssmJson!=null){
```

Comment:
Let us be consistent with this if null check

Reply:
I believe the ssmJson has the null check as expected. I didn't make any updates to the specified codeblock. If other changes are required please let me know.